tests/di_tests.py

```diff
@@ -310,6 +310,19 @@ class Subject:
         dm[Ham] = None
         subject.ham | should.be_None
 
+class DependencyMapProxyTests(unittest.TestCase):
+
+    def test_acts_as_proxy(self):
+        dm = ProxyDependencyMap()
+        dm[Ham] = Ham()
+        dm[Spam] = Spam()
+
+        ham = dm(Ham)
+        spam = dm(Spam)
+
+        ham | should.be_a(Ham)
+        spam | should.be_a(Spam)
```

Comment:
while the code is very similar to the one in the *werkzeug* project and it should be already battle tested, could you please add some specific tests for the basic functionality?

I'm thinking for instance testing what happens when we proxy a dependency that doesn't exist. Also one showcasing the modification of the proxied value would be cool for documentation purposes :memo: